internal/namespaces/baremetal/v1alpha1/custom_server_create.go

```diff
@@ -0,0 +1,118 @@
+package baremetal
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/scaleway/scaleway-cli/internal/core"
+	baremetal "github.com/scaleway/scaleway-sdk-go/api/baremetal/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func serverCreateBuilder(c *core.Command) *core.Command {
+	type baremetalCreateServerRequestCustom struct {
+		Zone scw.Zone `json:"-"`
+		// OrganizationID with which the server will be created
+		OrganizationID string `json:"organization_id"`
+		// Name of the server (≠hostname)
+		Name string `json:"name"`
+		// Description associated to the server, max 255 characters
+		Description string `json:"description"`
+		// Tags associated with the server
+		Tags []string `json:"tags"`
+		// Type of the server
+		Type string
+	}
+
+	c.ArgsType = reflect.TypeOf(baremetalCreateServerRequestCustom{})
+
+	c.ArgSpecs.DeleteByName("offer-id")
+
+	c.ArgSpecs.GetByName("name").Default = core.RandomValueGenerator("bm")
+	c.ArgSpecs.GetByName("description").Required = false
+
+	c.ArgSpecs.AddBefore("tags.{index}", &core.ArgSpec{
+		Name:    "type",
+		Short:   "Server commercial type",
+		Default: core.DefaultValueSetter("GP-BM1-M"),
+
+		EnumValues: []string{
+			// General Purpose offers
+			"GP-BM1-L",
+			"GP-BM1-M",
+
+			// High-computing offers
+			"HC-BM1-L",
+			"HC-BM1-S",
+
+			// High-Memory offers
+			"HM-BM1-XL",
+			"HM-BM1-M",
+		},
+	})
+
+	c.Run = func(ctx context.Context, argsI interface{}) (i interface{}, e error) {
+		client := core.ExtractClient(ctx)
+		api := baremetal.NewAPI(client)
+
+		tmpRequest := argsI.(*baremetalCreateServerRequestCustom)
```

Comment:
no request is made using `tmpRequest`

why not have a `createServerRequestCustom` where we add the field `OfferName` ? 

that way, `createServerRequestCustom` can be used to construct `baremetal.GetOfferFromOfferNameRequest` and `baremetal.CreateServerRequest`